### PR TITLE
[java] Add generics example

### PR DESCRIPTION
**Pull request by chwarr**

### CODE REVIEW

File: examples/java/core/generics/src/main/java/com/microsoft/bond/examples/Generics.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                Struct.BOND_TYPE.makeGenericType(
+                    BondTypes.STRING,
+                    BondTypes.DOUBLE,
+                    Base.BOND_TYPE.makeGenericType(BondTypes.WSTRING));
+
+        // Since Struct is generic, when constructing it, a type descriptor
+        // must be passed explicitly.
+        //
+        // Here, we pass a previously created type descriptor. Type
+        // descriptors are cached, so Struct.BOND_TYPE>makeGenericType() could
```
Comment (chadwalters) [inline examples/java/core/generics/src/main/java/com/microsoft/bond/examples/Generics.java]: Minor: Typo ('.' missing before "makeGenericType()")

File: examples/java/core/generics/src/main/java/com/microsoft/bond/examples/Generics.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        //
+        // Here, we pass a previously created type descriptor. Type
+        // descriptors are cached, so Struct.BOND_TYPE.makeGenericType() could
+        // be invoked directly in the constructor parameter list if wanted.
+        final Struct<String, Double, Base<String>> obj = new Struct<>(structBondType);
+        obj.n.x = "test";
+        obj.x = 0.11;
+        obj.y = "test";
+        obj.items.add(3.14);
+        obj.items.add(0D);
```
Comment (sigansky) [inline examples/java/core/generics/src/main/java/com/microsoft/bond/examples/Generics.java]: >D [](start = 23, length = 1)

Do you need "D"? It should do an implicit cast.

Comment (chwarr) [inline examples/java/core/generics/src/main/java/com/microsoft/bond/examples/Generics.java]: I'll check.

**Comment by chwarr** [comment]
One of the recent merges to the [jvm](https://github.com/Microsoft/bond/tree/jvm) branch have broken the C# build. I'll work on a fix on Tuesday, Pacific time.

File: examples/java/core/generics/src/main/java/com/microsoft/bond/examples/Generics.java (hunk 2 above)
Comment (chwarr) [inline examples/java/core/generics/src/main/java/com/microsoft/bond/examples/Generics.java]: The `D` is needed:

```
Generics.java:47: error: no suitable method found for add(int)
        obj.items.add(0);
                 ^
    method Collection.add(Double) is not applicable
      (argument mismatch; int cannot be converted to Double)
    method List.add(Double) is not applicable
      (argument mismatch; int cannot be converted to Double)
1 error
```

**Comment by chwarr** [comment]
Fix for cs.sln project is ready for review:
https://github.com/Microsoft/bond/pull/563

**Pull request merged**